### ISSUE TITLE
Sending same payment twice

### ISSUE THREAD

**Issue by btcontract** [ACINQ/eclair#128]
Currently Eclair won't accept a payment with used preimage (since related hash is deleted from `PaymentHandler` which results in `UnknownPaymentHash` on a second try) but it's possible to send the same payment twice by using the same payment request for example.  

Isn't this problematic? For example, in a payment between nodes `A` -> `B` -> `C`, `B` may remember a preimage on a first pass and then immediately fulfill a second identical payment without propagating it further.

**Comment by pm47** [comment]
You are correct that re-using payment requests is bad. Are you suggesting we remember paid payment hashes to mitigate this?

It is worth noting that the user will still be able to pay the same payment hash from another wallet and there is nothing we can do in that case.

**Comment by btcontract** [comment]
I've stumbled upon this while working on my own wallet, found no protection measures in Eclair and thought maybe I'm missing something and this is a non-issue, so decided to ask.

Turns out it is an issue in general but maybe not that much in Eclair since as far as I understand it's aimed at professional users who know what they're doing. I'll add protective measures in my wallet, however. Thanks for confirming my suspicions!
